src/legacy/core_plugins/kibana/public/doc_viewer/doc_viewer_tab.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { DocViewRenderProps, DocViewRenderFn } from 'ui/registry/doc_views';
+import { DocViewRenderTab } from './doc_viewer_render_tab';
+import { DocViewerError } from './doc_viewer_render_error';
+
+interface Props {
+  component?: React.Component | React.FunctionComponent;
+  id: number;
+  render?: DocViewRenderFn;
+  renderProps: DocViewRenderProps;
+  title: string;
+}
+
+interface State {
+  error: null | Error | string;
+  hasError: boolean;
+}
+/**
+ * Renders the tab content of a doc view.
+ * Displays an error message when it encounters exceptions, thank's to
+ * Error Boundaries.
+ */
+export class DocViewerTab extends React.Component<Props, State> {
+  state = {
+    hasError: false,
+    error: null,
+  };
+
+  static getDerivedStateFromError(error: unknown) {
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true, error };
+  }
+
+  shouldComponentUpdate(nextProps: Props, nextState: State) {
+    return (
+      nextProps.renderProps.hit._id !== this.props.renderProps.hit._id ||
+      nextProps.id !== this.props.id ||
+      nextState.hasError
+    );
+  }
+
+  render() {
+    const { component, render, renderProps, title } = this.props;
+    const { hasError, error } = this.state;
+
+    if (hasError && error) {
+      return <DocViewerError error={error} />;
+    } else if (!render && !component) {
+      return (
+        <DocViewerError
+          error={`Invalid plugin ${title}, there is neither a (react) component nor a render function provided`}
+        />
+      );
+    }
+
+    if (render) {
+      // doc view is provided by a render function, e.g. for legacy Angular code
+      return <DocViewRenderTab render={render} renderProps={renderProps} />;
+    }
+
+    // doc view is provided by a react component
+    const Component = component;
+    // @ts-ignore
+    return <Component {...renderProps} />;
```

Comment:
You could do

```jsx
const Component = component!;
return <Component {...renderProps} />;
```

---

At App Architecture team we agreed to not use `@ts-ignore` when the problem can be solved by casting, you can always cast a variable to `any`, which is better than TS-ignoring a whole line.

Reply:
tried that, no I've the following typescript complain

<img width="746" alt="Bildschirmfoto 2019-08-19 um 15 04 07" src="https://user-images.githubusercontent.com/463851/63267731-fdc3eb80-c292-11e9-82f5-f65246e499fa.png">

Reply:
ah wait a minute, I should have scrolled :)

Reply:
We've solved it by 

```
const Component = component as any;

```